pkg/services/govmomi/vcenter/clone.go

```diff
@@ -179,7 +169,6 @@ func Clone(ctx *context.VMContext, bootstrapData []byte, format bootstrapv1.Form
 			// the VM's UUID.
 			InstanceUuid:      string(ctx.VSphereVM.UID),
 			Flags:             newVMFlagInfo(),
-			DeviceChange:      deviceSpecs,
 			ExtraConfig:       extraConfig,
```

Comment:
Thanks for the PR! Code change looks good to me, only one question here, could you please help to explain why we are removing `DeviceChange`?

Reply:
We are not :) It has just been moved to the end https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/2134/files/950f4f01fd3345532424e0ee966b697ade43c8b4#diff-f96a0a333f63e93dee3e49d8995ee7bc623e31ded19b4c4172b880b58dc3af9aR292

The movement was because we don't know yet what's going to be the target datastore for the cloned device, so we can only mutate the datastore device spec once we know where is this going to live

Reply:
@rikatz thanks for the clarification, I missed that part